packages/modules/web_themes/koala/source/src/stores/mqtt-store.ts

```diff
@@ -2367,13 +2442,21 @@ export const useMqttStore = defineStore('mqtt', () => {
   ) => {
     return computed({
       get() {
-        const plans = vehicleScheduledChargingPlans(chargePointId).value;
+        const plans = vehicleScheduledChargingPlans.value(chargePointId);
         const plan = plans.find((p) => p.id === planId);
         return plan?.phases_to_use_pv;
       },
       set(newValue: number) {
-        const baseTopic = `openWB/chargepoint/${chargePointId}/set/charge_template/chargemode/scheduled_charging/plans/${planId}`;
-        updateTopic(baseTopic, newValue, 'phases_to_use_pv', true);
+        const plans = vehicleScheduledChargingPlans.value(chargePointId);
+        const planIndex = plans.findIndex((plan) => plan.id === planId);
+        if (planIndex === -1) return;
+        const objectPath = `chargemode.scheduled_charging.plans.${planIndex}.phases_to_use_pv`;
+        updateTopic(
+          `openWB/chargepoint/${chargePointId}/set/charge_template`,
+          newValue,
+          objectPath,
+          true,
+        );
```

Comment:
Hier wird mehrfach der gleiche Code ausgeführt. Kann man das in eine Methode zusammen fassen?